src/common/deflate.cpp

```diff
@@ -81,6 +81,63 @@ byte *decompressDeflate(const byte *data, size_t inputSize,
 	return decompressedData.release();
 }
 
+byte *decompressDeflate2(const byte *data, size_t inputSize, size_t &outputSize,
+						int windowBits, unsigned int frameSize) {
+	/* Initialize the zlib data stream for decompression with our input data.
+	 *
+	 * This ugly const cast is necessary because the zlib API wants a non-const
+	 * next_in pointer by default. Unless we define ZLIB_CONST, but that only
+	 * appeared in zlib 1.2.5.3. Not really worth bumping our required zlib
+	 * version for, IMHO. */
+
+	z_stream strm;
+	strm.zalloc   = Z_NULL;
+	strm.zfree    = Z_NULL;
+	strm.opaque   = Z_NULL;
+	strm.avail_in = inputSize;
+	strm.next_in  = const_cast<byte *>(data);
```

Comment:
Is it possible to pull all the inflate() stuff into its own common function, to minimize code duplication?

Reply:
Or at least the initializing...